paddle/fluid/operators/mkldnn/fc_mkldnn_op.cc

```diff
@@ -16,384 +16,126 @@ limitations under the License. */
-                                       const Tensor* weights) {
-    auto dims = Get4DWeightDimsForDNNL(input, weights);
-    return CreateMemDescriptor<float>(dims, MKLDNNMemoryFormat::oihw);
-  }
-
-  // Convert data from one data format to another
-  std::shared_ptr<mkldnn::memory> Reorder(const memory::desc& src_desc,
-                                          const memory::desc& dst_desc,
-                                          void* src_data) {
-    auto src_mem = memory(src_desc, engine_, src_data);
-    auto dst_mem = std::make_shared<memory>(dst_desc, engine_);
-
-    auto reorder = mkldnn::reorder(src_mem, *dst_mem);
-    auto& astream = platform::MKLDNNDeviceContext::tls().get_stream();
-
-    {
-      platform::RecordEvent record_reorder("int_reorder",
-                                           platform::EventRole::kUniqueOp);
-      reorder.execute(astream, src_mem, *dst_mem);
-      astream.wait();
+  void HandlePostOps(const paddle::framework::ExecutionContext& ctx,
+                     dnnl::primitive_attr& attrs) {
+    static std::unordered_map<std::string, dnnl::algorithm> algo_map = {
+        {"relu", dnnl::algorithm::eltwise_relu},
+        {"gelu", dnnl::algorithm::eltwise_gelu},
+        {"gelu_tanh", dnnl::algorithm::eltwise_gelu_tanh},
+        {"gelu_erf", dnnl::algorithm::eltwise_gelu_erf},
+        {"tanh", dnnl::algorithm::eltwise_tanh},
+        {"sigmoid", dnnl::algorithm::eltwise_logistic},
+        {"hard_swish", dnnl::algorithm::eltwise_hardswish},
+    };
+
+    if (IsInt8<T_w>()) {
+      const auto& output_shift_scale = ComputeOutputShiftScale(ctx);
+      int mask = CreateMask(1, output_shift_scale.size() > 1);
+      attrs.set_output_scales(mask, output_shift_scale);
     }
 
-    return dst_mem;
-  }
+    std::string activation_type = ctx.Attr<std::string>("activation_type");
 
-  // Convert data from one data format to another and rescale it.
-  // If the desired data type is (un)signed int8, quantization occurs here.
-  std::shared_ptr<mkldnn::memory> ReorderWithScale(
-      const std::shared_ptr<memory> src_mem, const memory::desc& dst_md,
-      const std::vector<float>& scale_data) {
-    auto dst_mem = std::make_shared<mkldnn::memory>(dst_md, engine_);
-    mkldnn::primitive_attr attributes;
-    // According to MKL-DNN's documentation mask determines along which
-    // dimensions should the scale be applied.
-    // 0 - Single scale applied to whole tensor
-    // 1 - Apply Scale along a slice of each dimension which index is 1.
-    //     In case of weights quantization, that dimension is output,
-    //     becuase we perform per-output-channel quantization
-    int mask = CreateMask(0, scale_data.size() > 1);
-    attributes.set_output_scales(mask, scale_data);
-    auto reorder = mkldnn::reorder(*src_mem, *dst_mem, attributes);
+    if (activation_type.empty() == false) {
+      constexpr float scale = 1.0f;
+      constexpr float alpha = 0.0f;
+      constexpr float beta = 0.0f;
```

Comment:
I can see in original code it is like that alpha = 1.0f and bet = 0.0f , but not sure if this is proper for gelu and some other activations. Could you please check with documentation if this is the way it should be?